### ISSUE TITLE
App does not work

### ISSUE THREAD

**Issue by Toxpox** [Erruar/Saku-Overclock#7]
Seems like program does not recognize the system. It freezes and closes after opening.
CPU: 7840HS 
Notebook: Acer Nitro AN16-41

[Debug_Report_28918247.8870617.txt](https://github.com/user-attachments/files/18243036/Debug_Report_28918247.8870617.txt)
![Ekran görüntüsü 2024-12-25 062123](https://github.com/user-attachments/assets/6411c77e-5591-4fd8-a5d8-cdac49033d35)

**Comment by Erruar** [comment]
I made a serious mistake in my app json in the latest version on github. I apologize for the inconvenience. I will release a new test version for everyone today. Just wait a little bit.

**Comment by Erruar** [comment]
Check new app version [there ](https://github.com/Erruar/Saku-Overclock/releases/tag/Saku-Overclock-1.0.18.31-Release-Candidate-9-CC31)

**Comment by Toxpox** [comment]
Now I can use program but with some issues.
UI is perfectly good.

Program still cannot recognize CPU. When I clicked to CPU Information section, program freezes and does not show any data.

Premade profiles limiting my CPU. I guess thats about recognition.
In Acer NitroSense;
Turbo Mode:
All Core Power Draw(peak//stable): 90W//60W
CPU TDC: 60+A
CPU EDC: 140A

When I choose Extreme premade preset in program max power draw around 30W. 
Biggest problem is when I did some changes in Parameters section computer is shuted down.

I really want to see it work.

**Comment by Erruar** [comment]
Premade profiles are just plain strings, they haven't auto detection right now, sorry, but I'll correct this in future. About Information page its really strange and its not OK. I'll look

**Comment by Toxpox** [comment]
![Ekran görüntüsü 2024-12-25 101926](https://github.com/user-attachments/assets/86a8811b-2b82-42c3-b18a-fb73bd958e8c)

![Ekran görüntüsü 2024-12-25 134202](https://github.com/user-attachments/assets/b2f1787c-dd4d-4938-a63e-bbd7fcb64de9)
When I approve my own settings, I receive a warning that all of them have been rejected. Through HWinfo, all values appear to be fixed and seem to have protection mode-like characteristics. For example, the package power does not exceed 25W, TDC is capped at a maximum of 25, EDC shows a maximum value of 55A, and the core voltage remains fixed at 0.88v. This includes IMC frequencies and other parameters as well. 

Am I missing something?

**Comment by Erruar** [comment]
# About error
![image](https://github.com/user-attachments/assets/6e1b1f44-d228-4fae-ad9c-876ffac9eabc)
its line 298, have you created your own profile?

# About fixed values
If you are running something vendor-specific software or any other Ryzen overclocking app, there can be a huge conflict that can lead to turning your laptop off, or fixed values as you see. I'd recommend to close any other software or set unlimited mode in them. When processor was unable to execute command, seems you need to reboot your laptop, or just try again. Seems something other is also using those commands now.

# If you want set more power
![image](https://github.com/user-attachments/assets/814e3f65-721c-42e0-99d1-47f6cd940373)
set cursor to any numberbox on Overclock page and set any value by keyboard or by mouse wheel down/up.
Dont forget to click on Apply button

**Comment by Toxpox** [comment]
Firstly, thanks for information.

Problem isn't your program. RyzenADJ is not working good for my PC. I can only effect TJmax and STAPM(--skin-temp-limit).

I finally access my BIOS and I realised that BIOS also doesn't have curve optimizer or frequency settings. Only thing the overclockable is RAM.
Also I mentioned about the CPU information section. Some times CPU info section is ignoring RyzenADJ and I can see WMI + PM Table data.  


I guess thats the end point for R7 7840HS.

**Comment by Erruar** [comment]
AFAIK skin temp limit isn't actual stapm, in my app its setting Laptop performance or overall performance, I haven't remembered, in Advanced section (with fire icon). Idk why info page isn't working for you, it should. I can't look right now, I'm in travelling. I'll look into it and say you all about fixes.

**Comment by Biotud** [comment]
I can run version 1.0.17.0 without any problems. With 1.0.18.31 it crashes as soon as I try to access Presets, Paramters, or Information.

I use it to try and fix the 0.4Ghz bug on a ryzen 3500u Lenovo E495 running Windows 11.

Also 1.0.17.0 doesn't remember not to check for updates when it has been closed, so I have to disable internet turn it off manually, start it again to prevent the auto-update notification.

**Comment by Erruar** [comment]
@Biotud does app gave you any errors or kind of that? Can you give me more information?
# About auto-update
I'll look into it and fix it disabling soon, wait for stable version

**Comment by Erruar** [comment]
@Toxpox does Ryzenadj -i command gave you something?

**Comment by Biotud** [comment]
> @Biotud does app gave you any errors or kind of that? Can you give me more information?
> 
> # About auto-update
> I'll look into it and fix it disabling soon, wait for stable version

Nope it just crashes, see attached video file :)

https://github.com/user-attachments/assets/2e81e9e9-7666-4d4e-9f08-68f8cbef1f59

**Comment by Erruar** [comment]
@Biotud can you manually download 1.0.18.0 version and check if all works. 1.0.18.31 its version only for test. Say me will 1.0.18.0 works for you

**Comment by Biotud** [comment]
@Erruar 

1.0.18.0 advised me that I needed to have Notebook FanControl installed to work, which I then did. I tried both 1.0.18.31 and 1.0.18.0 and as you can see 1.0.18.0 doesn't crash immediately, but it does crash. 

But maybe it is just this computer that isn't fully functional, being and older laptop :/  

https://github.com/user-attachments/assets/b3697cc4-d943-40b7-98a2-c7f4ba480bfa

**Comment by Erruar** [comment]
@Biotud on 18.0 only information page is crashing? And only if you enter on it second time?

**Comment by Biotud** [comment]
@Erruar 

I held up longer this tim I just pushed the different buttons several times for maybe 1-2 minutes changed power plan, then minimized it to tray and when I maximized it crashed again.

17.0 has yet to crash, it just bugs me for an update :P

**Comment by Erruar** [comment]
Since version 1.0.18.0 my application is build up with .NET 9.0.101, instead of .NET 7.0 as it was before and have completely different tray menu. Almost all application code was rewritten and a lot of things changed. That can cause bugs. Thanks for your watching. I'll introduce new update soon

**Comment by Biotud** [comment]
No problem, if there's a crash log or anything that can help you narrow down the problem, just let me know :)

**Comment by Erruar** [comment]
Sorry, since .NET 9.0 update they removed any message boxes and now no way to catch errors without manual code running in IDE. I'll introduce new error logging system... Soon.

**Comment by Toxpox** [comment]
Hi @Erruar ,
First of all; sorry for late response. 

RyzenADJ was working well. I could see information and change STAPM, EDC, TDC...

Problem was not in the apps. I found a way to reach Advanced BIOS and I saw nothing for CPU OC. All I found were settings related to power limits. 

I tried changing Pstate, and the laptop got bricked. (That's why I am responding late.)

In conclusion, overclocking the CPU with the R7 7840HS is a real headache. The best thing you can do is optimize the limits to achieve a perfect balance and avoid thermal throttling.

You can only change BCLK. (btw,  it doesn't seem to affect performance for me// no performance gains)

If I switch to an overclockable computer, I'll come back and bombard you with questions again. 😅

Thanks for your support.

**Comment by Erruar** [comment]
@Toxpox 

> I tried changing Pstate

... You shouldn't do that. There are one important thing. You can set freq, but with voltage... There is a trick. In Advanced BIOS it's not showing but in BIOS files there are sign: "Voltage for P states are setting in uV! 1V = 1000uV." Or something like that. If you set something like 1 or 2, in uV its 0.001V, thats why it got bricked. Its fixable, but only with physical flasher. Or in specialized service, it will cost maximum 50$ for your situation.

**Comment by Toxpox** [comment]
You are right, I shouldn't do that. Pstate voltage value was not decimal and its offset value. I changed it but I can't remember what I did. 

It was under warranty, so I sent it to the service center. I'm not sure if they'll figure out what I did. I could have done it myself with a CH341A and a 1.8V adapter, to be honest, it would have been better, but I didn’t want to bother.

If they check the current BIOS, it seems like it will be voided from the warranty.
